src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -176,29 +208,68 @@ class PalantirDockerPlugin implements Plugin<Project> {
         if (ext.pull) {
             buildCommandLine.add '--pull'
         }
-        buildCommandLine.addAll(['-t', "${ -> ext.name}", '.'])
+        buildCommandLine.addAll(['-t', "${-> ext.name}", '.'])
         return buildCommandLine
     }
 
+    @Deprecated
     private static String computeName(String name, String tag) {
-        int lastColon = name.lastIndexOf(':')
-        int lastSlash = name.lastIndexOf('/')
+        int firstAt = tag.indexOf("@")
 
-        int endIndex;
+        String tagValue
+        if (firstAt > 0) {
+            tagValue = tag.substring(firstAt + 1, tag.length())
+        } else {
+            tagValue = tag
+        }
 
-        // image_name -> this should remain
-        // host:port/image_name -> this should remain.
-        // host:port/image_name:v1 -> v1 should be replaced
-        if (lastColon > lastSlash) endIndex = lastColon
-        else endIndex = name.length()
+        if (tagValue.contains(':') || tagValue.contains('/')) {
+            // tag with ':' or '/' -> force use the tag value
+            return tagValue
+        } else {
+            // tag without ':' and '/' -> replace the tag part of original name
+            int lastColon = name.lastIndexOf(':')
+            int lastSlash = name.lastIndexOf('/')
 
-        return name.substring(0, endIndex) + ":" + tag
-    }
+            int endIndex;
+
+            // image_name -> this should remain
+            // host:port/image_name -> this should remain.
+            // host:port/image_name:v1 -> v1 should be replaced
+            if (lastColon > lastSlash) endIndex = lastColon
+            else endIndex = name.length()
 
-    private static String ucfirst(String str) {
-        StringBuffer sb = new StringBuffer(str);
-        sb.replace(0, 1, str.substring(0, 1).toUpperCase());
-        return sb.toString();
+            return name.substring(0, endIndex) + ":" + tagValue
+        }
     }
 
+    @Deprecated
+    private static String generateTagTaskName(String name) {
+        String tagTaskName = name
+        int firstAt = name.indexOf("@")
+
+        if (firstAt > 0) {
+            // Get substring of task name
+            tagTaskName = name.substring(0, firstAt)
+        } else if (firstAt == 0) {
+            // Task name must not be empty
+            throw new GradleException("Task name of docker tag '${name}' must not be empty.\n" +
+                    "There are some sample:\n" +
+                    "firstVersion@1.0.0 (dockerTagFirstVersion)\n" +
+                    "newName@docker-name:latest (dockerTagNewName)\n" +
+                    "withRepo@myregistryhost/fedora/httpd:version1.0 (dockerTagWithRepo)")
```

Comment:
I'd like to avoid specific examples in the source code. I think it's sufficient here to just have a clear error message.

The README is probably a better place for these examples.

Reply:
Got it.